### PR TITLE
Various improvements

### PR DESCRIPTION
**Pull request by stevescavo**
- Upgrade to Webpack 4 (https://github.com/MadeInHaus/react-redux-webpack-starter/issues/26)
- Improve/unify typography approach via a `Text` component
- Import normalize.css via npm for easier version bumping and abstraction
- Remove `react-css-modules` (closes https://github.com/MadeInHaus/react-redux-webpack-starter/issues/27)
- Remove Husky pre-commit hooks as they were buggy when bouncing between local npm installs, GUI-based git committing, and Docker deploys (https://github.com/MadeInHaus/react-redux-webpack-starter/issues/28)
- Add more presentation layer to suggest React patterns and support a sane typography approach
- Replace `babel-preset-es2015` with `babel-preset-env` (closes https://github.com/MadeInHaus/react-redux-webpack-starter/issues/29)
- Remove `react-waypoint` and `immutable` as they aren't appropriate for all project use cases

### CODE REVIEW

File: src/components/atoms/Button/Button.jsx [hunk 1]
```diff
@@ -1,25 +1,34 @@
...
 
-import './Button.scss';
+import styles from './Button.scss';
 
-const Button = ({ to, href, primary, ...props }) => {
-    const styleNames = cx('root', {
-        primary,
+const Button = ({ className, href, theme, to, ...props }) => {
+    const classNames = cx(styles.root, {
+        [className]: className,
```
Comment (unruffledBeaver) [inline src/components/atoms/Button/Button.jsx]: `className` can be moved out of the object and next to `styles.root`. As well, personally, I dont think we should allow classes to be passed in _unless_ they are mapped to classes that live in the component's style sheet.

File: src/assets/style-utils/colors.scss [hunk 2]
```diff
@@ -0,0 +1,51 @@
+// Colors
+// -------------------------
+// Need to define the raw colors somewhere!
+// stylelint-disable color-no-hex, function-max-empty-lines, value-list-max-empty-lines
+$color-palette-data: (
+    white: (base: #fff),
+    black: (base: #000),
+    gray: (base: #444, light: #666, dark: #222)
```
Comment (unruffledBeaver) [inline src/assets/style-utils/colors.scss]: If there are say 6 shades here, how do we name them?

File: src/components/atoms/Text/Text.jsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+            {children}
+        </TagName>
+    );
+};
+
+Text.propTypes = {
+    children: PropTypes.node.isRequired,
+    className: PropTypes.string,
+    tag: PropTypes.string,
+    theme: PropTypes.string,
```
Comment (unruffledBeaver) [inline src/components/atoms/Text/Text.jsx]: This should probably be a oneOf PropType to throw a warning when someone passes a bad theme

File: src/components/atoms/Text/Text.jsx [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+
+import styles from './Text.scss';
+
+const Text = ({ children, className, tag, theme, ...props }) => {
+    const TagName = tag;
+    const classNames = cx({
+        [className]: className,
+        [styles.headline]: theme === 'headline',
```
Comment (unruffledBeaver) [inline src/components/atoms/Text/Text.jsx]: If these are mutually exclusive visual states, meaning 2 cannot be applied at once, we should only apply the theme once. Something like: `const themeClass = styles[theme];`

File: src/components/atoms/Button/Button.jsx (hunk 1 above)
Comment (stevescavo) [reply src/components/atoms/Button/Button.jsx]: > As well, personally, I dont think we should allow classes to be passed in unless they are mapped to classes that live in the component's style sheet.

What's your argument? I don't see a problem with passing custom classes to components as needed. Building layout wrappers around every ui component seems a bit too purist for my taste and leads to overly verbose markup.

File: src/assets/style-utils/colors.scss (hunk 2 above)
Comment (stevescavo) [reply src/assets/style-utils/colors.scss]: Up to the author, but these typically are best derived from what's been agreed upon in the Zeplin stylesheet.

File: src/components/atoms/Button/Button.jsx (hunk 1 above)
Comment (unruffledBeaver) [reply src/components/atoms/Button/Button.jsx]: Its my opinion, but I have found it easier to deal with styles on a global level when components are only concerned with their own visual states. It opens the door for people to abuse the "theming" pattern and for collisions. Also using wrappers over passing in classes has made it a lot easier to replace components in layout in my experience. 

Whats wrong with being purist? I'd rather have a little extra markup over hunting down a style collision.

File: src/assets/style-utils/colors.scss (hunk 2 above)
Comment (stevescavo) [reply src/assets/style-utils/colors.scss]: Suggestion?

Comment (unruffledBeaver) [reply src/assets/style-utils/colors.scss]: I use a plugin that calls this api http://chir.ag/projects/ntc/

If we, as devs are responsible for naming the colors, this takes the thought process out and theyre all unique. Maybe this is a requirement from creative though?

Comment (stevescavo) [reply src/assets/style-utils/colors.scss]: At HAUS, we're not responsible for naming colors. We work with the designers to name them (or take their lead), establish them in the Zeplin style guide, and then transfer them to our codebase.

File: src/components/atoms/Button/Button.jsx (hunk 1 above)
Comment (stevescavo) [reply src/components/atoms/Button/Button.jsx]: I prefer being fairly pragmatic when approaching custom, visually complex interfaces. You can be a purist in a code sample/Medium article/codepen, but that mindset doesn't lend itself to hacking together a great experience on a relatively tight timeline. Creating pure layout wrappers for _everything_ is too rigid of an assumption in an edge-case quest to *maybe* avoid style collisions. We're already using locally-scoped styles via css modules, so as long as we're diligent about *not* styling type selectors, we should very rarely (if ever) have style collisions.

File: src/components/atoms/Text/Text.jsx (hunk 3 above)
Comment (stevescavo) [reply src/components/atoms/Text/Text.jsx]: Good call. Updated: https://github.com/MadeInHaus/react-redux-webpack-starter/pull/34/commits/c523efb98ced812945754e885fbedfebf2c49eff

File: src/components/atoms/Text/Text.jsx (hunk 4 above)
Comment (stevescavo) [reply src/components/atoms/Text/Text.jsx]: Another good call. Updated: https://github.com/MadeInHaus/react-redux-webpack-starter/pull/34/commits/457f670fc40d827710fda0125ca53a20210f5a9e

**Comment by stevescavo** [comment]
Appreciate the feedback @unruffledBeaver 👍

**Pull request merged**